mypy/checker.py

```diff
@@ -468,23 +474,25 @@ def get_generator_yield_type(self, return_type: Type, is_coroutine: bool) -> Typ
             # parameters (i.e. is `object`), then the yielded values can't
             # be accessed so any type is acceptable.  IOW, ty is Any.
             # (However, see https://github.com/python/mypy/issues/1933)
-            return AnyType()
+            # todo: is type of this Any correct?
+            return AnyType(TypeOfAny.special_form)
```

Comment:
It's not clear to me that this should even return `Any`, but I think special form is fine here.